### PR TITLE
Add test for applicable host search URI

### PR DESCRIPTION
**Pull request by swadeley**
Closed loop: BZ#1838800

Hello

this PR adds a test for:
  [BZ#1838800 – Manage Errata from Content Host Page does not provide link to view list of content hosts affected by an Errata.](https://bugzilla.redhat.com/show_bug.cgi?id=1838800)

it depends on:
[ Add search_applicable_hosts #662 ](https://github.com/SatelliteQE/airgun/pull/662)

### CODE REVIEW

**Comment by swadeley** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py::test_positive_end_to_end_bulk_update
https://github.com/SatelliteQE/airgun/pull/662

File: tests/foreman/ui/test_contenthost.py [hunk 1]
```diff
@@ -246,6 +247,9 @@ def test_positive_end_to_end_bulk_update(session, default_location, vm):
     hc_name = gen_string('alpha')
     description = gen_string('alpha')
     result = vm.run(f'yum -y install {FAKE_1_CUSTOM_PACKAGE}')
+    search_uri = (
+        f'{default_sat.hostname}/content_hosts?search=installable_errata%3D{FAKE_1_ERRATA_ID}'
```
Comment (jyejare) [inline tests/foreman/ui/test_contenthost.py:251]:
```suggestion
        f'{default_sat.hostname}/content_hosts?search=installable_errata={FAKE_1_ERRATA_ID}'
```
??

**Comment by swadeley** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py::test_positive_end_to_end_bulk_update
airgun: 662

File: tests/foreman/ui/test_contenthost.py [hunk 2]
```diff
@@ -262,23 +266,29 @@ def test_positive_end_to_end_bulk_update(session, default_location, vm):
...
         # Wait for upload profile event (in case Satellite system slow)
-        host = entities.Host().search(query={'search': f'name={vm.hostname}'})
         wait_for_tasks(
-            search_query='label = Actions::Katello::Host::UploadProfiles'
-            f' and resource_id = {host[0].id}'
-            f' and started_at >= "{timestamp}"',
+            search_query='label = Actions::Katello::Applicability::Hosts::BulkGenerate'
+            f' and started_at >= "{timestamp}"'
+            f' and state = stopped'
+            f' and result = success',
```
Comment (jyejare) [inline tests/foreman/ui/test_contenthost.py:291]: Why its been moved from `host tasks to wait for` to `a collection of hosts` tasks, is this applies to more than one host ?

**Comment by swadeley** [comment]
> Overall looks good with a curious question and a comment!
> 
> Also, I see the test has become a big enough and failing in one assert will cause other steps to skip/fail for long until we fix the asserts, so instead why dont we have a separate test for search URI ?

I thought it was such a small check and wanted to save test execution time by adding it to an existing test. Lets see what other say.

**Review by pondrejk — APPROVED** [review]

**Review by omkarkhatavkar — CHANGES_REQUESTED** [review]

File: tests/foreman/ui/test_contenthost.py (hunk 2 above)
Comment (swadeley) [reply tests/foreman/ui/test_contenthost.py:291]: Hi @jyejare 

`Actions::Katello::Host::UploadProfiles` does not exist in Sat7

see:
 Change applicability task check #9323

File: tests/foreman/ui/test_contenthost.py [hunk 3]
```diff
@@ -262,23 +266,29 @@ def test_positive_end_to_end_bulk_update(session, default_location, vm):
...
-        # make a note of time for later CLI wait_for_tasks, and include
-        # 8 mins margin of safety.
-        timestamp = (datetime.utcnow() - timedelta(minutes=8)).strftime('%Y-%m-%d %H:%M')
+        # For BZ#1838800, assert the Host Collection Errata Install table has the search URI
+        uri = session.hostcollection.search_applicable_hosts(hc_name, FAKE_1_ERRATA_ID)
+        assert search_uri in uri
+        # Ensure Errata dialog is closed
+        session.browser.click(
+            locator="(//div[contains(@class, 'modal-header')]/.//button[@ng-click='cancel()'])"
+        )
```
[the remark this answers is not in the document]
Comment (omkarkhatavkar) [reply tests/foreman/ui/test_contenthost.py:275]: you might be running with a time synchronization issue, for me, both worked well 

view.dialog.cancel()
view.dialog.confirm()

dont use direct click rather wrapper functions.

**Comment by swadeley** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py::test_positive_end_to_end_bulk_update
airgun: 662

**Review by omkarkhatavkar — APPROVED** [review]

**Comment by swadeley** [comment]
robottelo]$ pytest --pdbcls pudb.debugger:Debugger --pdb tests/foreman/ui/test_contenthost.py::test_positive_end_to_end_bulk_update
============================== test session starts ===============================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo_2/robottelo, configfile: pyproject.toml
plugins: ibutsu-2.0.2, mock-3.6.1, xdist-2.5.0, forked-1.3.0, services-2.2.1, cov-3.0.0, reportportal-5.0.8
collected 1 item                                                                                                                                                                                  

tests/foreman/ui/test_contenthost.py .                                                                                                                                                      [100%]

=========================== 1 passed in 1015.61s (0:16:55) ========================================

**Comment by swadeley** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py::test_positive_end_to_end_bulk_update

**Comment by swadeley** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py::test_positive_end_to_end_bulk_update
airgun: 662

File: tests/foreman/ui/test_contenthost.py (hunk 1 above)
Comment (jyejare) [reply tests/foreman/ui/test_contenthost.py:251]: @swadeley ??

Comment (swadeley) [reply tests/foreman/ui/test_contenthost.py]: OK, changed that, and rebased

**Comment by swadeley** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py::test_positive_end_to_end_bulk_update

**Review by jyejare — APPROVED** [review]

**Comment by jyejare** [comment]
@swadeley Did you look into PRT failure and it isnt for this PR?  Let me know so we can merge it !!

**Comment by swadeley** [comment]
@jyejare PRT build 656 has failed due to :
  
    16:04:12  >       raise exception_class(message, screen, stacktrace)

    16:04:12  E       selenium.common.exceptions.WebDriverException: Message: Error forwarding the new session Request timed out waiting for a node to become available.

**Comment by swadeley** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py::test_positive_end_to_end_bulk_update

**Comment by jyejare** [comment]
@swadeley The test is failing in repository-set enablement which doesnt looks like the issue with test but probably manifest / CDN.

Merging your PR based on the test results you provided above.

**Comment by swadeley** [comment]
Thank you

**Pull request merged**